AirportTicketBookingExercise/Logic/Handlers/Command/PassengerCommandHandler.cs

```
@@ -0,0 +1,237 @@
+            else
+            {
+                FilterParam SearchParam = productInfo[1].ParseFilterParam();
+                string input = productInfo[2];
+
+                var flights = _flightService.GetFlights();
+            filteredFlights = SearchParam switch
+                {
+                    FilterParam.Flight => flights.Where(f => f.FlightName.Equals(input)).ToList(),
+                    FilterParam.Price => flights.Where(f =>
+                        f.BuisnessPrice == decimal.Parse(input) ||
+                        f.EconomyPrice == decimal.Parse(input) ||
+                        f.FirstClassPrice == decimal.Parse(input)).ToList(),
+                    FilterParam.DepartureCountry => flights.Where(f => f.DepartureCountry.Equals(input)).ToList(),
+                    FilterParam.DestinationCountry => flights.Where(f => f.DestinationCountry.Equals(input)).ToList(),
+                    FilterParam.DepartureDate => flights.Where(f => f.DepartureDate.Equals(input)).ToList(),
+                    FilterParam.DepartureAirport => flights.Where(f => f.DepartureAirport.Equals(input)).ToList(),
+                    FilterParam.ArrivalAirport => flights.Where(f => f.ArrivalAirport.Equals(input)).ToList(),
+                    _ => []
+                };
+            }
+            return filteredFlights;
+        }
+
+        public bool Cancel(string[] productInfo, User loggedInUser)
+        {
+            try
+            {
+                int bookingId = int.Parse(productInfo[1]);
+                if (!_bookingService.IsBookingValidById(bookingId, loggedInUser.UserId))
+                    return false;
+                else
+                {
+                    bool isSuccess = _bookingService.RemoveBookingById(bookingId);
+                    if (!isSuccess)
+                        return false;
+                    else return true;
+                }
+            }
+            catch
+            {
+                return false;
```

we return false in multiple paths here, each one for a different reason, we shouldn't swallow exception and just return false, better to throw an  excpetion with the proper message for each path that we think is failing.

---

re-review the PR and see where we return false and if we should throw exceptions instead